packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -485,8 +485,48 @@ class TimeInput extends React.Component {
     if ((event.key === '-' || event.key === '_') && !this.props.atMinDate) {
       const currentTimeValue = this.formatHour(hour, meridiem).concat(':', minute).concat(this.props.showSeconds ? ':'.concat(second) : '');
       if (TimeUtil.validateTime(currentTimeValue, this.props.showSeconds)) {
-        if (minute === '0' || minute === '00') {
-          minute = '59';
+        if (inputType === 3 && this.props.showSeconds) {
+          if ((second === '0' || second === '00')) {
+            second = '59';
+            if (minute === '0' || minute === '00') {
+              minute = '59';
+              if (hour === '0' || hour === '00') {
+                hour = '23';
+              } else {
+                if (variant === TimeUtil.FORMAT_12_HOUR && hour === '12') {
+                  if (meridiem === this.anteMeridiem) {
+                    meridiem = this.postMeridiem;
+                  } else {
+                    meridiem = this.anteMeridiem;
+                  }
+                }
+                hour = TimeUtil.decrementHour(hour, variant);
+              }
+            } else {
+              minute = TimeUtil.decrementMinute(minute);
+            }
+          } else {
+            second = TimeUtil.decrementSecond(second);
+          }
+        } else if (inputType === 1) {
+          if (minute === '0' || minute === '00') {
+            minute = '59';
+            if (hour === '0' || hour === '00') {
+              hour = '23';
+            } else {
+              if (variant === TimeUtil.FORMAT_12_HOUR && hour === '12') {
+                if (meridiem === this.anteMeridiem) {
+                  meridiem = this.postMeridiem;
+                } else {
+                  meridiem = this.anteMeridiem;
+                }
+              }
+              hour = TimeUtil.decrementHour(hour, variant);
+            }
+          } else {
+            minute = TimeUtil.decrementMinute(minute);
+          }
+        } else if (inputType === 0) {
```

Comment:
Can you simplify the if else block

Reply:
Updated - https://github.com/cerner/terra-framework/pull/1966/commits/4caa69c649e337786153c7767f908c3a07d605e7